lib/_io.scm

```diff
@@ -8344,6 +8344,143 @@
 
 ;;;----------------------------------------------------------------------------
 
+(define-prim (##make-raw-device-port raw-device device uname direction)
+  (let ((mutex
+         (macro-make-port-mutex))
+        (rkind
+         (if (##eq? direction (macro-direction-out))
+           (macro-none-kind)
+           (macro-raw-device-kind)))
+        (wkind
+         (if (##eq? direction (macro-direction-in))
+           (macro-none-kind)
+           (macro-raw-device-kind)))
+        (roptions
+         0)
+        (rtimeout
+         #t)
+        (rtimeout-thunk
+         #f)
+        (woptions
+         0)
+        (wtimeout
+         #t)
+        (wtimeout-thunk
+         #f)
+        (rdevice-condvar
+         (and (##not (##eq? direction (macro-direction-out)))
+              (##make-rdevice-condvar raw-device)))
+        (wdevice-condvar
+         (and (##not (##eq? direction (macro-direction-in)))
+              (##make-wdevice-condvar raw-device)))
+        (uname (or uname 'raw-device)))
+
+    (define (name port)
+
+      ;; It is assumed that the thread **does not** have exclusive
+      ;; access to the port.
+
+      (##declare (not interrupts-enabled))
+      (##list uname (macro-raw-device-port-device port)))
```

Comment:
I use it as a means for typing raw devices, and also having a nice display of the port.
I could call it name and make it non-optional, so that we are consistent and don't default in raw-device.
Finally, I can see the point about non-closures, I can simply add the name (better, type) to the `raw-device-port` type.